pennylane/operation.py

```diff
@@ -465,6 +465,12 @@ def generator(self):
         """
         return [None, 1]
 
+    @staticmethod
```

Comment:
It is actually very smart to have this as a static method, but I also like to call
```python
U3(a, b, c, wires=[2]).decomposition()
```
directly, not having to pass arguments that are already contained in the object `U3`. This could be used to simplify the code in the QNode afterwards a little bit.

This might be too much, but what do you think of adding a  non-static method `decomposition_template` (probably to the base class `Operation`) that calls `decomposition` with the unpacked parameters of the instance and the corresponding wires?

Reply:
This would be nice, but I have a gut feeling this will not work because of queuing --- U3 would end up being added alongside the decomposition. You could have logic that removes U3 from the queue first, but I feel this starts to get a bit too complex?

Note that as currently implemented, you don't need to pass the parameters twice, but you pass it as follows:

```python
U3.decomposition(a, b, c, wires=[2])
```

Hmm let me think about this.

Reply:
Yeah, I know. The staticmethod enables using the decomposition it in a context where queuing is happening. I was just thinking about the QNode where you have an `Operation` object and you could just call `op.decomposition()` instead of `op.decomposition(*op.params, wires=op.wires)`. But it's not much of a hassle anyways, because it will most likely just be used inside of the `QNode` and not in the plugins.